android/java/org/chromium/chrome/browser/app/domain/KeyringModel.java

```diff
@@ -264,38 +279,59 @@ public void getAccounts(Callbacks.Callback1<AccountInfo[]> callback1) {
 
     public void addAccount(String accountName, @CoinType.EnumType int coinType,
             Callbacks.Callback1<Boolean> callback) {
-        final AccountInfo[] finalAccountInfos =
+        final AccountInfo[] finalAccountInfo =
                 WalletUtils.getAccountInfosFromKeyrings(_mKeyringInfosLiveData.getValue())
                         .toArray(new AccountInfo[0]);
-        mKeyringService.addAccount(accountName, coinType, result -> {
-            if (result) {
-                boolean hasExistingAccountType = false;
-                for (AccountInfo accountInfo : finalAccountInfos) {
-                    hasExistingAccountType = (accountInfo.coin == coinType);
-                    if (hasExistingAccountType) break;
-                }
-                if (!hasExistingAccountType) {
-                    mKeyringService.getKeyringInfo(
-                            getSelectedCoinKeyringId(coinType), updatedKeyringInfo -> {
-                                for (AccountInfo accountInfo : updatedKeyringInfo.accountInfos) {
-                                    if (accountInfo.coin == coinType) {
-                                        setSelectedAccount(accountInfo.address, coinType);
-                                        break;
-                                    }
-                                }
-                            });
-                }
-            }
-            mCryptoSharedActions.updateCoinType();
-            mCryptoSharedActions.onNewAccountAdded();
-            callback.call(result);
-        });
+        if (coinType == CoinType.FIL) {
+            mKeyringService.addFilecoinAccount(
+                    accountName, BraveWalletConstants.FILECOIN_MAINNET, result -> {
```

Comment:
This method is being used as many different places so other than `AddAccountActivity`, we should pass the selected Filecoin network.